File: 49-group-anagrams/49-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
        unordered_map<string, vector<string>> table;
        vector<vector<string>> res;
        
        for(int i = 0; i < strs.size(); i++){
            
            string temp = strs[i];
            sort(temp.begin(), temp.end());
            table[temp].push_back(strs[i]);
            
        }
        
        for(auto item : table){
            
            res.push_back(item.second);
        }
        
        return res;
    }
};

File: 49-group-anagrams/49-group-anagrams.go
func groupAnagrams(strs []string) [][]string {
    
    m := make(map[[26]int][]string)
    
    for _, item := range strs {
        var table [26]int
        for _, char := range item {
            
            table[char - 'a']++
        }
        m[table] = append(m[table], item)
    }
    
   var res [][]string
    
    for _, val := range m {
        res = append(res, val)   
    }
    
    return res
}